### DISCUSSION TITLE
Feature request: Add in_stock filter to catalog price rules?

### DISCUSSION THREAD

**Discussion by seansan** [OpenMage/magento-lts#2126] [Ideas]
### Description (*)
Price rules always costs a lot of time and resources; to improve its speed we thought about adding an in_stock filter (for us 80% of all products are out of stock and old products)

### Expected behavior (*)
Add filter option to catalog price rules products in_stock = yes/no
Difficulty here is that the rules work for parent products ... and also has an option to change child products, so stock should be checked at the correct level (salable product level)

### Benefits
Speed, reduces load by X% - 80% - at least in our case

### Additional information
![image](https://user-images.githubusercontent.com/652395/91457291-7a18e380-e884-11ea-95ca-a32e3b829186.png)

**Comment by Flyingmana** [comment]
so it would be an additional condition which could be added to rules to improve performance?
or would this be something which is done in the background?

First would be great, because its complete backwards compatible

**Comment by seansan** [comment]
I would aim at 1st solution ....

bg not smart, because there is always the odd situation where stock levels
can fluctuate during the day and the sore owner needs the products that
come back into stock to be discounted too
